src/_guides/language/concurrency/index.md

```diff
@@ -0,0 +1,456 @@
+
+
+### Event handling
+
+In a client app, the main isolate’s event queue might contain
+repaint requests and notifications of tap and other UI events, like this:
+***[PENDING: describe the figure in more detail.]***
+
+![PENDING: alt text here](/guides/language/concurrency/images/event-loop.png)
+
+The event handling happens on the main isolate after `main()` exits:
+***[PENDING: describe the figure in more detail.]***
+
+![PENDING: alt text here](/guides/language/concurrency/images/event-handling.png)
+
+If a synchronous operation takes too much processing time,
+the app can become unresponsive.
+In the following figure, the tap-handling code takes too long,
+so subsequent events are handled too late:
+***[PENDING: describe the figure in more detail.]***
+
+![PENDING: alt text here](/guides/language/concurrency/images/event-jank.png)
+
+In client apps, the result of a too-lengthy synchronous operation is often
+[janky (non-smooth) UI animation][jank].
+Worse, the UI might become completely unresponsive.
+
+[jank]: https://docs.flutter.dev/perf/rendering
+
+
+### Background workers
+
+If your app’s UI becomes unresponsive due to a time-consuming computation —
+[parsing a large JSON file][json], for example —
+consider offloading that computation to a worker isolate,
+often called a _background worker._
+A common case, shown in the following figure,
+is spawning a simple worker isolate that
+performs a computation and then exits.
+The worker isolate returns its result in a message when the worker exits.
+
+[json]: https://docs.flutter.dev/cookbook/networking/background-parsing)
+
+![PENDING: alt text here](/guides/language/concurrency/images/isolate-bg-worker.png)
+<br>
+***[PENDING: Inside the figure, Background worker -> Worker isolate]***
+
+Each isolate message can deliver one object,
+which includes anything that’s transitively reachable from that object.
+Not all object types are sendable, and
+the send fails if any transitively reachable object is unsendable.
+For example, you can send an object of type `List<Object>` only if
+none of the objects in the list is unsendable.
+If one of the objects is, say, a `Socket`, then
+the send fails because sockets are unsendable.
+
+For information on the kinds of objects that you can send in messages,
+see the API reference documentation for the [`send()` method][].
+
+A worker isolate can perform I/O
+(reading and writing files, for example), set timers, and more.
+It has its own memory and
+doesn’t share any state with the main isolate.
+The worker isolate can block without affecting other isolates.
+
+[`send()` method]: https://api.dart.dev/stable/dart-isolate/SendPort/send.html
+
+
+## Code examples
+
+This section discusses some examples
+that use the `Isolate` API
+to implement isolates.
+
+{{site.alert.tip}}
+  If you're using Flutter on a non-web platform,
+  then instead of using the `Isolate` API directly,
+  consider using the [Flutter `compute()` function][].
```

Comment:
Can we extend this with something like ", which is a convenience function for moving a single function call to a worker isolate" ?